### PR TITLE
improve http header

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: src/providers/generic_http.jl [hunk 1]
```diff
@@ -16,16 +16,28 @@ const bundle_key = AssetRegistry.register(normpath(joinpath(
...
 Serve an asset from the asset registry.
 """
 function serve_assets(req)
     if haskey(AssetRegistry.registry, req.target)
         filepath = AssetRegistry.registry[req.target]
         if isfile(filepath)
+            mime = get(
+                known_mimetypes,
+                extension(filepath),
+                "application/octet-stream"
```
Comment (pfitzseb) [inline src/providers/generic_http.jl]: Is this really a good fallback? Or could we just not send a `"Content-Type"` if it can't be determined?

File: src/providers/generic_http.jl [hunk 2]
```diff
@@ -16,16 +16,28 @@ const bundle_key = AssetRegistry.register(normpath(joinpath(
...
+            )
             return HTTP.Response(
                 200,
-                [],
+                [
+                    "Access-Control-Allow-Origin" => "*",
```
Comment (pfitzseb) [inline src/providers/generic_http.jl]: AFAIU it's a "best practice" not to set this to `"*"`, but  ¯\\\_(ツ)\_/¯ I guess :P

Comment (SimonDanisch) [reply src/providers/generic_http.jl]: Oh... well maybe it's not needed anyways :D

Comment (travigd) [reply src/providers/generic_http.jl]: Actually, I'm not sure that this is necessary at all. `GET` requests that don't include [strange headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Simple_requests) shouldn't trigger CORS at all.

File: src/providers/generic_http.jl [hunk 3]
```diff
@@ -16,16 +16,27 @@ const bundle_key = AssetRegistry.register(normpath(joinpath(
...
+const requests = []
 """
 Serve an asset from the asset registry.
 """
 function serve_assets(req)
+    push!(requests, req)
     if haskey(AssetRegistry.registry, req.target)
         filepath = AssetRegistry.registry[req.target]
         if isfile(filepath)
+            headers = ["Access-Control-Allow-Origin" => "*"]
```
Comment (travigd) [inline src/providers/generic_http.jl]: I think this section would read better if you defined a `filename2mime` style method (or `ext2mime` or whatever) and then simply defined headers as
```
headers = [
    "Access-Control-Allow-Origin" => "*",
    "Content-Type" => filepath2mime(filepath),
]
```

Not a deal breaker either way.

File: src/providers/generic_http.jl (hunk 2 above)
Comment (travigd) [reply src/providers/generic_http.jl]: If CORS is being triggered, then maybe SystemJS adds strange things to the headers that aren't allowed under the specification for _simple requests_.

File: src/providers/generic_http.jl (hunk 1 above)
Comment (travigd) [reply src/providers/generic_http.jl]: `application/octet-stream` is a good fallback IMO. It's the default (according to the HTTP spec) if the content type isn't specified and can't be guessed (by inspecting the file and using whatever heuristics the browser has implemented).

Source: http://mattryall.net/blog/2008/03/default-content-type

File: src/providers/generic_http.jl (hunk 2 above)
Comment (travigd) [reply src/providers/generic_http.jl]: Oh! Sorry for the spam, I was mistaken. _Simple requests_ don't trigger **preflight** requests (_i.e._ `OPTIONS` requests that precede the real request), but they still subscribe to CORS rules about whether or not the client (the originator of the XHR) can access the content.

tl;dr: If we're only using (_simple_) `GET`'s then we **don't** need to worry about `OPTIONS` requests but **do** need to set the `Access-Control-Allow-Origin` header (and I think in this case, `*` is acceptable).

So what @SimonDanisch has is correct.

File: src/providers/generic_http.jl (hunk 1 above)
Comment (pfitzseb) [reply src/providers/generic_http.jl]: Ah okay, didn't I kinda assumed HTTP would default to something else. Carry on :P
